feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/presentation/referenda/common/RealReferendumFormatter.kt

```diff
@@ -0,0 +1,402 @@
+import kotlin.time.Duration.Companion.days
+import kotlin.time.Duration.Companion.milliseconds
+
+private val oneDay = 1.days
+
+private data class AccountVoteFormatComponent(
+    val direction: VoteDirectionModel,
+    val amount: String,
+    val votes: String,
+    val multiplier: String
+)
+
+class RealReferendumFormatter(
+    private val resourceManager: ResourceManager,
+    private val trackFormatter: TrackFormatter,
+    private val referendaStatusFormatter: ReferendaStatusFormatter
+) : ReferendumFormatter {
+
+    override fun formatVoting(voting: ReferendumVoting, token: Token): ReferendumVotingModel {
+        return ReferendumVotingModel(
+            positiveFraction = voting.approval.ayeVotesIfNotEmpty()?.fraction?.toFloat(),
+            thresholdFraction = voting.approval.threshold.value.toFloat(),
+            votingResultIcon = R.drawable.ic_close,
+            votingResultIconColor = R.color.icon_negative,
+            thresholdInfo = formatThresholdInfo(voting.support, token),
+            thresholdInfoVisible = !voting.support.passing(),
+            positivePercentage = resourceManager.getString(
+                R.string.referendum_aye_format,
+                voting.approval.ayeVotes.fraction.formatFractionAsPercentage()
+            ),
+            negativePercentage = resourceManager.getString(
+                R.string.referendum_nay_format,
+                voting.approval.nayVotes.fraction.formatFractionAsPercentage()
+            ),
+            thresholdPercentage = resourceManager.getString(
+                R.string.referendum_to_pass_format,
+                voting.approval.threshold.value.formatFractionAsPercentage()
+            )
+        )
+    }
+
+    override fun formatReferendumTrack(track: ReferendumTrack, asset: Chain.Asset): ReferendumTrackModel {
+        val trackModel = trackFormatter.formatTrack(track.track, asset)
+
+        return ReferendumTrackModel(trackModel, sameWithOther = track.sameWithOther)
+    }
+
+    override fun formatOnChainName(call: GenericCall.Instance): String {
+        return "${call.module.name}.${call.function.name}"
+    }
+
+    override fun formatUnknownReferendumTitle(referendumId: ReferendumId): String {
+        return resourceManager.getString(R.string.referendum_name_unknown, formatId(referendumId))
+    }
+
+    private fun formatThresholdInfo(
+        support: ReferendumVoting.Support,
+        token: Token
+    ): String {
+        val thresholdFormatted = mapAmountToAmountModel(support.threshold.value, token).token
+        val turnoutFormatted = token.amountFromPlanks(support.turnout).format()
+
+        return resourceManager.getString(R.string.referendum_support_threshold_format, turnoutFormatted, thresholdFormatted)
+    }
+
+    override fun formatStatus(status: ReferendumStatus): ReferendumStatusModel {
+        val statusName = referendaStatusFormatter.formatStatus(status.type)
```

Comment:
It's not showing in diff: I added formatting status name separately and use it below